ddspipe_core/CMakeLists.txt

```diff
@@ -70,9 +70,20 @@ compile_library(
 # Test
 ###############################################################################
 # Compile tests if CMake options requires it
-compile_test_library(
-    "${PROJECT_SOURCE_DIR}/test" # Test directory
-)
+
+# Determine Fast DDS version
+if ("${fastrtps_VERSION}" VERSION_LESS 2.13)
+    set(DDS_TYPES_VERSION "/v2/")
+else()
+    set(DDS_TYPES_VERSION "/v1/")
+endif()
+
+file(GLOB_RECURSE TEST_FILES "${PROJECT_SOURCE_DIR}/test/")
+foreach(TEST_FILE ${TEST_FILES})
+    if(NOT TEST_FILE MATCHES ${DDS_TYPES_VERSION})
+    compile_test_library(${TEST_FILE})
+    endif()
+endforeach()
```

Comment:
This has the same issue as we already resolved for the router. You are setting test files while the function request the path to tests.